Documentation/GeneralConventions/FileStructure.rst

```diff
@@ -0,0 +1,810 @@
+^^^^^^^^^^^^^^
+
+If the *use_opensearch* property is set to the public URL of the documentation,
+Sphinx renders an OpenSearch description file of the documentation, which allows
+most browsers to treat the internal search of your documentation as a global
+search engine and include it in their search bar like any other search engine.
+This seems useful if your documentation is a large reference that is of high
+public interest and likely to be searched frequently by readers. Good examples
+of this are the PHP or TYPO3 Core references, for example
+"https://docs.typo3.org/m/typo3/reference-typoscript/main/en-us/", while the
+TYPO3 extension documentations probably are not.
+
+
+.. _settings-cfg-intersphinx-mapping:
+
+Intersphinx mapping
+^^^^^^^^^^^^^^^^^^^
+
+The *[intersphinx_mapping]* section contains base URLs of public manuals for
+convenient and verified cross-referencing.
+
+For example, if you uncomment the *t3start* mapping in the Settings.cfg above,
+the :samp:`https://docs.typo3.org/m/typo3/tutorial-getting-started/main/en-us/Extensions/Management.html`
+page can be referenced with
+
+.. code-block:: rst
+
+   :doc:`t3start:Extensions/Management`
+
+and the fragment :samp:`#install-extension-with-composer` on the same page with
+
+.. code-block:: rst
+
+   :ref:`t3start:install-extension-with-composer`
+
+The prerequisite is that the target manual is also compiled with Sphinx and
+provides an objects.inv file in the root of the documentation that
+contains all reference targets. Since this file is binary, the rendering
+toolchain provides a twin objects.inv.json file that allows the reader
+to easily look up reference targets. An example of this is this
+`production objects.inv.json <https://docs.typo3.org/m/typo3/tutorial-getting-started/main/en-us/objects.inv.json>`__.
+
+If a target reference becomes unavailable during documentation rendering, for
+example if it has been permanently removed, a warning is issued and stored in
+the toolchain log file at :file:`Documentation-GENERATED-temp/Result/project/0.0.0/_buildinfo/warnings.txt`.
+
+
+.. _single-file-documentation:
+
+Single file documentation
+=========================
+
+This setup is not recommended, but can be handy for those who want to publish
+their documentation on docs.typo3.org and want to keep their mono README.md
+documentation style for now.
+
+This structure allows the author to minimize effort by maintaining a single
+documentation file that is interpreted simultaneously by the VCS host
+and the Sphinx theme. On the other hand, the author only has
+the reduced set of content elements supported by the former, rather than using
+the theme's rich selection of :ref:`custom content elements <rest-reference>`.
+
+Of course, you can also use a README.rst instead of a README.md file if you
+prefer its syntax.
+
+.. code-block:: none
+
+   .
+   ├── README.md
+   └── Documentation
+       └── Settings.cfg
+
+
+.. _single-file-documentation-readme-md:
+
+README.md
```

Comment:
We are using rst all over. Why should we suggest mark down here even though rst is possible?

Reply:
Yes, as the documentation is also for TYPO3 extension documentations, where many like to keep the README.md, i think it is fine to provide examples for both cases here.

Reply:
I would prefer to suggest to new comers to use rst and then just mention (and give an example) that you can also keep the README.md instead of rst.

Reply:
Ok, will invert the order.

Reply:
Done.